File: mediq-frontend/src/services/patientService.js
import api from './api';

export const patientService = {
  registerPatient: async (data) => api.post('/patients', data),
  updatePatient: async (id, data) => api.put(`/patients/${id}`, data),
  getPatientById: async (id) => api.get(`/patients/${id}`),
  getPatientByPatientId: async (patientId) => api.get(`/patients/patient-id/${patientId}`),
  getPatientsByPhone: async (phone) => api.get(`/patients/phone/${phone}`),
  searchPatients: async (campId, keyword, page = 0, size = 10) =>
    api.get('/patients', { params: { campId, keyword, page, size } }),
  getPatientTimeline: async (id) => api.get(`/patients/${id}/timeline`),
  deletePatient: async (id) => api.delete(`/patients/${id}`),
};

export default patientService;

File: mediq-backend/src/main/java/com/mediq/service/PatientService.java
package com.mediq.service;

import com.mediq.dto.PatientResponse;
import com.mediq.dto.RegisterPatientRequest;
import com.mediq.dto.UpdatePatientRequest;
import org.springframework.data.domain.Page;

import java.util.List;

public interface PatientService {

    PatientResponse registerPatient(RegisterPatientRequest request);

    PatientResponse updatePatient(Long patientId, UpdatePatientRequest request);

    PatientResponse getPatientById(Long patientId);

    PatientResponse getPatientByPatientId(String patientId);

    List<PatientResponse> getPatientsByPhone(String phone);

    Page<PatientResponse> searchPatients(Long campId, String keyword, int page, int size, String sortBy, String sortDir);

    void deletePatient(Long patientId);
}
